DigitalLearningSolutions.Web/ViewComponents/NumericInputViewComponent.cs

```diff
@@ -0,0 +1,39 @@
+﻿namespace DigitalLearningSolutions.Web.ViewComponents
+{
+    using System.Linq;
+    using DigitalLearningSolutions.Web.ViewModels.Common.ViewComponents.DigitalLearningSolutions.Web.ViewModels.Common.
+        ViewComponents;
+    using Microsoft.AspNetCore.Mvc;
+
+    public class NumericInputViewComponent : ViewComponent
+    {
+        public IViewComponentResult Invoke(
+            object model,
+            string aspFor,
+            string label,
+            bool populateWithCurrentValue,
+            string type,
+            string hintText,
+            string cssClass
+        )
+        {
+            var property = model.GetType().GetProperty(aspFor);
+            var valueToSet = populateWithCurrentValue ? property?.GetValue(model)?.ToString() : null;
+
+            var errorMessages = ViewData.ModelState[property?.Name]?.Errors.Select(e => e.ErrorMessage) ??
+                                new string[] { };
+
+            var numberViewModel = new NumericInputViewModel(
+                aspFor,
+                aspFor,
+                label,
+                valueToSet,
+                type,
+                errorMessages,
+                string.IsNullOrEmpty(cssClass) ? null : cssClass,
+                string.IsNullOrEmpty(hintText) ? null : hintText
+            );
+            return View("~/Views/Shared/Components/NumericInput/Default.cshtml", numberViewModel);

```

Comment:
Does specifying "Default" not work on its own?

Reply:
Doesn't seem to, no. I see that all the view components with configurable models use the entire string. I created the NumericInputViewComponent from the TextInputWithConfigurableModelViewComponent, so I've renamed it to NumericInputWithConfigurableModelViewComponent to reflect that.

Reply:
Ah I imagine it's to do with the naming. I've had a look, and each of those xWithConfigurableModelViewComponent use the same ViewModel and View as the regular view component without the configurable model. But seeing as how those suggested paths are automatically determined from the names, it's probably looking for a view called "Default" in a folder call "NumericInputWithConfigurableModel" which doesn't exist as we only have the "NumericInput" folder. So having consistency across each file/folder/class name would probably solve the issue.

HOWEVER, as part of https://softwiretech.atlassian.net/browse/HEEDLS-613 we are getting rid of the "xWithConfigurableModel" view components, so its probably best just to rename this "NumericInputViewCompontent". The view string here will probably accept "Default" after that, but if it doesn't then that's probably fine.

Reply:
Cool, named it back to NumericInputViewComponent, which let me remove the view file string entirely like most other view components :)